doc/mclag/MCLAG_Enhancements_HLD.md

```diff
@@ -0,0 +1,805 @@
+8. MclagSyncd updates MAC addresses learned from peer MCLAG node to new  MCLAG FDB table .
+9. FdbOrch registers for new MCLAG FDB table updates to process MAC updates from peer MCLAG node, ISOGRP Orch process new updates from MclagSyncd.
+10. FdbOrch updates remote MAC addresses with new SAI attribute not to age them out. 
+11. Syncd gets the ICCP learned remote MAC addresses with new SAI attribute SAI_FDB_ENTRY_TYPE_STATIC_MACMOVE set .
+12. Syncd programs the remote MAC to HW
+14. FdbOrch update the MAC address to STATE_DB MCLAG REMOTE FDB Table.
+15. FdbSycd gets the MAC updates from STATE_DB MCLAG REMOTE FDB Table.
+16. FdbSyncd programs the MAC updates to kernel.
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+#### 3.2.1.1 MCLAG Domain table
+
+Producer: Configuration
+Consumer: MclagSyncd 
+Description: New table to store MCLAG domain configuration. Applications can use it to get MCLAG configuration
+
+```
+;New MCLAG Domain Table
+key               = MCLAG_DOMAIN|domain_id ; DIGIT 1-4095 
+
+;field             = value
+source_ip          = IP               ; local ip address 
+peer_ip            = IP               ; peer  ip address 
+peer_link          = port_name        ; peer link name
+keepalive_interval = time_in_secs     ; keepalive time in seconds 
+session_timeout    = time_in_secs     ; session timeout in multiple of 
+									  ; keepalive_interval.
+
+;value annotations
+port_name         = 1*64VCHAR        ; name of the port, must be unique
+time_in_secs      = dec-octet        ; time in seconds
+IP                = IPV4address
+IPv4address        = dec-octet "." dec-octet "." dec-octet "." dec-octet "
+
+ dec-octet        = DIGIT            ; 0-9
+                / %x31-39 DIGIT      ; 10-99
+                / "1" 2DIGIT         ; 100-199
+                / "2" %x30-34 DIGIT  ; 200-249
+                / "25" %x30-35       ; 250-255
+```
+#### 3.2.1.2 MCLAG Interface Table
+
+Producer: Configuration
+Consumer: MclagSyncd 
+Description: New table to store MCLAG interfaces for a MCLAG domain. MCLAG applications can query this table to check whether a interface is configured as MCLAG interface
+
+```
+;New MCLAG Interface Table
+key              = MCLAG_INTERFACE|domain_id|ifname ; 
+
+;value annotations
+domain_id 	   = DIGIT 1-4095     ; domain_id for the MCLAG
+ifname         = 1*64VCHAR        ; name of the MCLAG Interface (PortChannel)
+
+```
+
+Note: Same PortChannel name (ifname) is required to configure across all the MCLAG peer nodes towards MHD. 
+
+#### 3.2.1.3 MCLAG UniqueIP Table
+
+Producer: Configuration
+Consumer: MclagSyncd 
+Description: New table to store L3 VLAN interfaces that will have unique IP configuration. MCLAG applications can query this table to check whether a interface is configured for L3 protocol support.
+
+```
+;New MCLAG UniqueIP Table
+key              = MCLAG_UNIQUEIP_TABLE|ifname ; Only VLAN interface supported currently
```

Comment:
I believe it's should be `MCLAG_UNIQUE_IP` as per https://github.com/Azure/sonic-swss-common/blob/b75aab7636021199d3f6498eb2295f8e9dc34d73/common/schema.h#L290

Reply:
Thanks @akokhan  , corrected the typo.